packages/std/src/testing/mock.rs

```diff
@@ -1428,6 +1477,33 @@ mod tests {
         );
     }
 
+    #[cfg(feature = "cosmwasm_1_3")]
+    #[test]
+    fn distribution_querier_delegator_withdraw_address() {
+        let mut distribution = DistributionQuerier::new(HashMap::new());
+        distribution.set_withdraw_addresses(
+            [("addr0".to_string(), "withdraw0".to_string())]
+                .into_iter()
+                .collect(),
+        );
```

Comment:
This usage looks cumbersome. We have `DistributionQuerier::default();` for creating an empty one. Not sure if we should implement Default though.

And then I suggest adding a `set_withdraw_address(delegator_address: impl Into<String>, withdraw_address:  impl Into<String>>` function to add a single entry.

I'm not sure if we need `set_withdraw_addresses`. This replaces all existing data which is misleading IMO. When I have some existing addresses set, why would `set_withdraw_addresses` clear them?

Reply:
I think we should implement Default. All the other mock queriers do and at least conceptually I think the querier should still work if we provide no info to it.
But I agree on the `set_withdraw_address` idea. That seems nicer.
I wanted some way to clear them because I wasn't sure if the distinction between "not set" and "set to the same address" is significant, but that is probably better solved with a `clear` method anyways.

Reply:
> I think we should implement Default.

👍

> because I wasn't sure if the distinction between "not set" and "set to the same address" is significant

In Cosmos SDK you cannot unset it once you set it
- https://github.com/cosmos/cosmos-sdk/blob/v0.47.3/proto/cosmos/distribution/v1beta1/tx.proto#L49-L60
- https://github.com/cosmos/cosmos-sdk/blob/v0.47.3/x/distribution/types/msg.go#L56-L58

So unsetting a single address is not important IMO. Resetting the whole thing might make sense in a longer test flow.